### PR TITLE
Fixes #30876 - Add permissions for host_collections controller

### PR DESCRIPTION
**Pull request by jeremylenz**

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -387,7 +387,7 @@ Style/InverseMethods:
   Enabled: false
 
 # Offense count: 2
-Style/MethodMissingSuper:
+Lint/MissingSuper:
```
Comment (jeremylenz) [inline .rubocop_todo.yml:390]: this fixes a Rubocop deprecation warning, figured it can't hurt to fix it here

Comment (jeremylenz) [reply .rubocop_todo.yml:390]: update: it hurt

**Comment by jeremylenz** [comment]
[test katello]

File: app/controllers/katello/api/v2/host_collections_controller.rb [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+    before_action :find_authorized_katello_resource, :only => [
+      :copy,
+      :show,
+      :update,
+      :destroy,
+      :add_hosts,
+      :remove_hosts,
+      :hosts
+    ]
+    before_action :find_readable_activation_key
```
Comment (parthaa) [inline app/controllers/katello/api/v2/host_collections_controller.rb:13]: I see this method only being used by index method.
```suggestion
    before_action :find_readable_activation_key, :only => [:index]
```

File: app/controllers/katello/api/v2/host_collections_controller.rb [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+    before_action :find_editable_host
```
Comment (parthaa) [inline app/controllers/katello/api/v2/host_collections_controller.rb:14]:
```suggestion
    before_action :find_editable_host, :only => [:index]
```

**Comment by parthaa** [comment]
Can you add a test similar to https://github.com/Katello/katello/blob/master/test/controllers/api/v2/sync_plans_controller_test.rb#L130-L137

**Comment by jeremylenz** [comment]
@parthaa thanks, updated!

**Review by parthaa — APPROVED** [review]

**Pull request merged**